analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/PipedInputScanner.kt

```diff
@@ -13,6 +13,7 @@ fun <R> InputStream.mapLines(transform: (String) -> R): List<R> {
 
 fun InputStream.forEachLine(action: (String) -> Unit) {
     val scanner = Scanner(this)
+    Thread.sleep(1000)
     if (available() <= 0) return // Otherwise it will get stuck waiting for user input
```

Comment:
So, as discussed I like the early returns :) Here it does not really matter since there are only few lines after the condition, but in general i think it is nicer. Here is an illustrative example from stackexchange, that shows why i think that:

```
public int SomeFunction(bool cond1, string name, int value, AuthInfo perms)
{
    int retval = SUCCESS;
    if (someCondition)
    {
        if (name != null && name != "")
        {
            if (value != 0)
            {
                if (perms.allow(name)
                {
                    // Do Something
                }
                else
                {
                    reval = PERM_DENY;
                }
            }
            else
            {
                retval = BAD_VALUE;
            }
        }
        else
        {
            retval = BAD_NAME;
        }
    }
    else
    {
        retval = BAD_COND;
    }
    return retval;
}
```

vs

```
public int SomeFunction(bool cond1, string name, int value, AuthInfo perms)
{
    if (!someCondition)
        return BAD_COND;

    if (name == null || name == "")
        return BAD_NAME;

    if (value == 0)
        return BAD_VALUE;

    if (!perms.allow(name))
        return PERM_DENY;

    // Do something
    return SUCCESS;
}
```